### DISCUSSION TITLE
Unable to authorize with private repository if the repository is blocked by a VPN

### DISCUSSION THREAD

**Discussion by ahatzz11** [renovatebot/renovate#16188] [Request Help] [answered]
👋 I'm using the [Renovate Github App](https://github.com/apps/renovate) and trying to configure a private nexus repository and am currently unable to get it to properly work. I'm attempting to do this with `hostRules` that match my domain, and a username with encrypted password. I'm using the [renovate encrypt tool](https://app.renovatebot.com/encrypt) for the encryption of my password, which I have replaced with `{encrypted-password-value}` below.

---

Currently our onboarding PRs have a bunch of this which is what I'm trying to resolve:
### ⚠ Dependency Lookup Warnings ⚠

Please correct - or verify that you can safely ignore - these lookup failures before you merge this PR.

-   `Failed to look up dependency com.example.infrastructure.libraries:date-util`

---

`renovate.json`:
```json
{
  "extends": [
    "config:base"
  ],
  "hostRules": [
    {
      "hostType": "maven",
      "matchHost": "example.com",
      "username": "renovate",
      "encrypted": {
        "password": "{encrypted-password-value}"
      }
    }
  ]
}
```

This is the migrated json:

```
DEBUG: migrated config
{
  "config": {
    "extends": [
      "github>whitesource/merge-confidence:beta",
      "config:base"
    ],
    "hostRules": [
      {
        "hostType": "maven",
        "matchHost": "example.com",
        "username": "renovate",
        "encrypted": {
          "password": "***********"
        }
      }
    ]
  }
}
```

In the logs for the renovate job I see that it successfully decrypted and it says authentication is being added for my domain:

```
DEBUG: Found encrypted config
{
  "config": {
    "password": "***********"
  }
}
DEBUG: Trying to decrypt password
DEBUG: Decrypted config using openpgp
DEBUG: Decrypted password
DEBUG: Setting hostRules from config
DEBUG: Adding password authentication for example.com to hostRules
```

Up until this point it seems like everything is properly setup, but once dependencies start being checked I get authentication errors:

```
DEBUG: Dependency lookup unauthorized. Please add authentication with a hostRule
{
  "failedUrl": "https://nexus.example.com/repository/example/com/example/infrastructure/libraries/date-util/maven-metadata.xml"
}
```

If I go to the `failedUrl` (at my own proper domain) I am able to see the file after authenticating. 

Here's some other stuff that I have tried:
- Using a password that has no special characters before the encryption
- Token auth with an encrypted value (didn't work, but I'd also prefer to use username/password)
- Changing my `matchHost` domain a bit, e.g. `https://nexus.example.com` and `example.com` and `https://nexus.example.com/repository/`, probably others
- Playing with `npmrc`
- Using `hostType` and omitting it

I'm having a hard time seeing in the logs what is going wrong, so I'm hoping that someone could throw some additional ideas my way to try, thanks in advance!

--- 
**Important Edit**: One thing that we've come up with internally is that our nexus repository sits behind our VPN. Do you have a list of IP addresses that the renovate bot runs from so we could include it in our allowlist for the VPN? I think we could fix this by self-hosting as well but if we could use the hosted version that would be awesome.

**Comment by rarkins** [comment, marked as answer]
As you mention in your edit, this will be caused if your registry is behind a VPN and/or allow list of IP addresses and Renovate can't reach it
